src/Presentation/Controllers/ExpenseController.cs

```diff
@@ -0,0 +1,115 @@
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using System.Security.Claims;
+using ExpenseTrackerGrupo4.src.Aplication.Interfaces;
+using ExpenseTrackerGrupo4.src.Presentation.DTOs;
+using AutoMapper;
+using ExpenseTrackerGrupo4.src.Domain.Entities;
+
+namespace ExpenseTrackerGrupo4.src.Presentation.Controllers;
+
+[ApiController]
+[Route("api/expenses")]
+[Authorize]
+public class ExpenseController(IExpenseService expenseService, IMapper mapper) : ControllerBase
+{
+    private readonly IExpenseService _expenseService = expenseService;
+    private readonly IMapper _mapper = mapper;
+
+    [HttpPost]
+    [Authorize]
+    public async Task<IActionResult> CreateExpense([FromBody] CreateUpdateExpenseDto dto)
+    {
+        var userId = GetCurrentUserId();
+
+        if(userId == Guid.Empty) return Forbid(); 
+
+        var expense = _mapper.Map<Expense>(dto);
+        expense.UserId = userId; 
+
+        await _expenseService.AddAsync(expense);
+        return CreatedAtAction(nameof(GetExpenseById), new { id = expense.Id }, dto);
+    }
+
+    [HttpGet]
+    [Authorize]
+    public async Task<IActionResult> GetUserExpenses(
+        [FromQuery] DateTime? startDate, [FromQuery] DateTime? endDate, [FromQuery] string? category
+    )
+    {
+        var userId = GetCurrentUserId();
+
+        if(userId == Guid.Empty) return Forbid(); 
+
+        var expenses = await _expenseService.GetUserExpensesCommand(userId, startDate, endDate, category);
+
+        return Ok(expenses);

```

Comment:
When a range of a day is set to itself, the answer is empty even if there is data for that date.
![image](https://github.com/user-attachments/assets/31fc53a3-456e-42b9-82fb-3348d7052f30)
![image](https://github.com/user-attachments/assets/d4014fe7-0774-4927-ad61-446230b5a941)